pkg/planner/core/plan_clone_generator.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"reflect"
+)
+
+// genPlanCloneForPlanCacheCode generates CloneForPlanCache method for all physical plan nodes.
+// Using code-gen is safer than writing by hand, for example, if someone adds a new field to a struct,
+// the code-gen can update the Clone method correctly and automatically.
```

Comment:
Is there any instruction on how to update it automatically?

Reply:
I added more comments, PTAL